api/src/utils/validate-uuid.ts

```diff
@@ -0,0 +1,6 @@
+// This expression is a bit more relaxed than libraries like uuid and uuid-validate - see: https://github.com/directus/directus/issues/21573
+const REGEX = /^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$/i;
```

Comment:
Instead of the current comment, I'd like to have an explanation of the pattern and why that is valid/works for our use-case? So the main difference is obviously that we allow any version. Is there another thing which is more relaxed?

Do you have a source for that RegEx pattern - just to make it easier for us to validate?

Reply:
> Instead of the current comment, I'd like to have an explanation of the pattern and why that is valid/works for our use-case. So the main difference is obviously that we allow any version. Is there another thing which is more relaxed?
> 
> Do you have a source for that RegEx pattern - just to make it easier for us to validate?
> 

I've included a more detailed comment on the pattern, including reference sources.

> Bonus: I know this works with PostgreSQL, but I couldn't find an official statement about non-standard versions (cannot see anything here [1](#user-content-fn-1-e6889ca2085226678b7bd91f114ce744)) - do you happen to have anything to hand?

As far as I am aware there is no official statement, however there is no validation performed (see [here](https://github.com/postgres/postgres/blob/master/src/backend/utils/adt/uuid.c)) and there are extensions (such as [pg_uuidv7](https://pgxn.org/dist/pg_uuidv7/)) that introduce methods to generate version 7 UUIDs.